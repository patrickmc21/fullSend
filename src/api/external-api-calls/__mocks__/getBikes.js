import { mockBikes } from '../../../mock-data/mock-data';

const getBikes = jest.fn().mockImplementation(() => {
  return Promise.resolve(mockBikes);
});

export default getBikes;