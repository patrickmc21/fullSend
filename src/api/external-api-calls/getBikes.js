const getBikes = async (bikeId, token) => {
  const url = `https://www.strava.com/api/v3/gear/${bikeId}`;
  const optionsObject = {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`
    }
  };

  try {
    const response = await fetch(url, optionsObject);
    const data = await response.json();
    return data;  
  } catch (error) {
    throw error.message;
  }
};

export default getBikes;