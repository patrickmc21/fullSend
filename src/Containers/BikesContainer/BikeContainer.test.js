import React from 'react';
import { shallow } from 'enzyme';
import * as mock from '../../mock-data/mock-data';
import { 
  BikeContainer, 
  mapStateToProps, 
  mapDispatchToProps 
} from './BikeContainer';

import getBikes from '../../api/external-api-calls/getBikes';
jest.mock('../../api/external-api-calls/getBikes');


describe('BikeContainer', () => {

  let mockUser;
  let mockBikes;
  let mockAddBikes;
  let wrapper;

  beforeEach(() => {
    mockUser = {...mock.mockUser,...mock.mockStravaUser};
    mockBikes = mock.mockBikes;
    mockAddBikes = jest.fn();
    wrapper = shallow(
      <BikeContainer
        user={mockUser}
        bikes={mockBikes}
        addBikes={mockAddBikes}/>,
      {disableLifeCycleMethods: true}
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getBikes on getUserBikes', async () => {
    const expected = [mockUser.bikes[0].id, mockUser.token];
    wrapper.instance().getUserBikes(mockUser.bikes, mockUser.token);
    expect(getBikes).toHaveBeenCalledWith(...expected);
  });

  it('should return an array of bikes on getUserBikes', async () => {
    const expected = mockBikes;
    const results = await wrapper.instance().getUserBikes(mockUser.bikes, mockUser.token);
    expect(results).toEqual([expected]);
  });

  it('should getUserBikes on mount if no bikes present', () => {
    mockBikes = [];
    wrapper = shallow(
      <BikeContainer
        user={mockUser}
        bikes={mockBikes}
        addBikes={mockAddBikes}/>,
      {disableLifeCycleMethods: true}
    )
    const spy = jest.spyOn(wrapper.instance(), 'getUserBikes');
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should call addBikes on mount if no bikes present', async () => {
    mockBikes = [];
    wrapper = shallow(
      <BikeContainer
        user={mockUser}
        bikes={mockBikes}
        addBikes={mockAddBikes}/>,
      {disableLifeCycleMethods: true}
    )
    await wrapper.instance().componentDidMount();
    expect(mockAddBikes).toHaveBeenCalled();
  });

});

describe('mapStateToProps', () => {

  let mockState;
  let mapped;

  beforeEach(() => {
    mockState = {
      user: mock.mockUser,
      bikes: mock.mockBikes
    };
    mapped = mapStateToProps(mockState);
  });

  it('should map user to props', () => {
    const expected = mock.mockUser;
    expect(mapped.user).toEqual(expected);
  });

  it('should map bikes to props', () => {
    const expected = mock.mockBikes;
    expect(mapped.bikes).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should map addBikes to props', () => {
    const mockDispatch = jest.fn();
    const expected = {
      type: 'ADD_BIKES',
      bikes: mock.mockBikes
    };
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addBikes(mock.mockBikes);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
