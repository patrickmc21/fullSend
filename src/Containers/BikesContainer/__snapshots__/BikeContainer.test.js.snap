// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BikeContainer
    addBikes={[Function]}
    bikes={
        Array [
            Object {
              "brand_name": "BMC",
              "description": "My Bike.",
              "distance": 388206,
              "frame_type": 3,
              "id": "b1231",
              "model_name": "Teammachine",
              "primary": false,
              "resource_state": 3,
            },
          ]
    }
    user={
        Object {
            "athlete_type": 1,
            "badge_type_id": 4,
            "bikes": Array [
              Object {
                "distance": 0,
                "id": "b12345678987655",
                "name": "EMC",
                "primary": true,
                "resource_state": 2,
              },
            ],
            "city": "San Francisco",
            "clubs": Array [],
            "country": "US",
            "created_at": "2017-11-14T02:30:05Z",
            "date_preference": "%m/%d/%Y",
            "email": "mv@abc.com",
            "firstname": "Marianne",
            "follower": null,
            "follower_count": 5,
            "friend": null,
            "friend_count": 5,
            "ftp": null,
            "id": 1234567890987654400,
            "lastname": "V.",
            "measurement_preference": "feet",
            "mutual_friend_count": 0,
            "name": "Pat",
            "premium": true,
            "profile": "https://xxxxx.cloudfront.net/pictures/athletes/123456789/123456789/2/large.jpg",
            "profile_medium": "https://xxxxxx.cloudfront.net/pictures/athletes/123456789/123456789/2/medium.jpg",
            "resource_state": 3,
            "sex": "F",
            "shoes": Array [
              Object {
                "distance": 4904,
                "id": "g12345678987655",
                "name": "adidas",
                "primary": true,
                "resource_state": 2,
              },
            ],
            "state": "CA",
            "token": "2y589wyasdfhgfiouqehr8to8",
            "updated_at": "2018-02-06T19:32:20Z",
            "username": "marianne_v",
            "weight": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="bike-card-container"
>
        <BikeCard
                bike={
                        Object {
                                "brand_name": "BMC",
                                "description": "My Bike.",
                                "distance": 388206,
                                "frame_type": 3,
                                "id": "b1231",
                                "model_name": "Teammachine",
                                "primary": false,
                                "resource_state": 3,
                              }
                }
        />
</div>,
      "className": "bike-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BikeCard
            bike={
                        Object {
                                    "brand_name": "BMC",
                                    "description": "My Bike.",
                                    "distance": 388206,
                                    "frame_type": 3,
                                    "id": "b1231",
                                    "model_name": "Teammachine",
                                    "primary": false,
                                    "resource_state": 3,
                                  }
            }
/>,
        ],
        "className": "bike-card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "b1231",
          "nodeType": "function",
          "props": Object {
            "bike": Object {
              "brand_name": "BMC",
              "description": "My Bike.",
              "distance": 388206,
              "frame_type": 3,
              "id": "b1231",
              "model_name": "Teammachine",
              "primary": false,
              "resource_state": 3,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bike-card-container"
>
          <BikeCard
                    bike={
                              Object {
                                        "brand_name": "BMC",
                                        "description": "My Bike.",
                                        "distance": 388206,
                                        "frame_type": 3,
                                        "id": "b1231",
                                        "model_name": "Teammachine",
                                        "primary": false,
                                        "resource_state": 3,
                                      }
                    }
          />
</div>,
        "className": "bike-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BikeCard
              bike={
                            Object {
                                          "brand_name": "BMC",
                                          "description": "My Bike.",
                                          "distance": 388206,
                                          "frame_type": 3,
                                          "id": "b1231",
                                          "model_name": "Teammachine",
                                          "primary": false,
                                          "resource_state": 3,
                                        }
              }
/>,
          ],
          "className": "bike-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "b1231",
            "nodeType": "function",
            "props": Object {
              "bike": Object {
                "brand_name": "BMC",
                "description": "My Bike.",
                "distance": 388206,
                "frame_type": 3,
                "id": "b1231",
                "model_name": "Teammachine",
                "primary": false,
                "resource_state": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifeCycleMethods": true,
  },
}
`;
